### PR TITLE
nydusd: fix race window in copen handler

### PR DESCRIPTION
**Pull request by kevinXYin**
we should make sure blob cache init thread get wlock before user daemon
get first ondemand req.

Fixes: 82f67123192c ("nydusd: make copen always return right file size")
Suggested-by: Jiang Liu <gerry@linux.alibaba.com>
Signed-off-by: Xin Yin <yinxin.x@bytedance.com>

### CODE REVIEW

**Comment by changweige** [comment]
Could you help elaborate further about the race window this PR is closing?

File: src/bin/nydusd/fs_cache.rs [hunk 1]
```diff
@@ -462,17 +462,21 @@ impl FsCacheHandler {
             e.insert((FsCacheObject::DataBlob(fsblob.clone()), msg.fd));
             state.id_to_config_map.insert(hdr.object_id, config.clone());
             let blob_size = config.blob_info().deref().uncompressed_size();
-            Self::init_blobcache(fsblob);
+            let barrier = Arc::new(Barrier::new(2));
+            Self::init_blobcache(fsblob, Arc::clone(&barrier));
+            // make sure blob cache init thread get wlock before get first ondemand req
+            barrier.wait();
```
Comment (changweige) [inline src/bin/nydusd/fs_cache.rs:468]: Nit: 
Some grammar mistakes  with this comment sentence 😹

Comment (kevinXYin) [reply src/bin/nydusd/fs_cache.rs:468]: fixed ~

**Comment by kevinXYin** [comment]
> Could you help elaborate further about the race window this PR is closing?

updated in commit message

**Review by jiangliu — APPROVED** [review]

**Pull request merged**